### ISSUE TITLE
chaincode has a problem with stub.CreateTable

### ISSUE THREAD

**Issue by novusopt** [hyperledger-archives/fabric#1049]
Hi,

currently i am looking at the development of chaincode and i tried to write a simple chaincode which creates a table in the Init callback function.

Some notes about my setup:
- 1 validating peer with security enabled & noops as consensus algorithm-> command to run the peer: `sudo CORE_SECURITY_ENABLED=true CORE_SECURITY_PRIVACY=true ./fabric peer --peer-chaincodedev`
- membership service is running
- command to deploy the chaincode: `CORE_CHAINCODE_ID_NAME=chaincodetest CORE_PEER_ADDRESS=127.0.0.1:30303 ./chaincodetest`
- command to call the init function: `./fabric chaincode deploy -n chaincodetest -u user1 -c  '{"Function":"init", "Args": ["arg1"]}'` -> user1 is logged in &  registered as a client

My Init function

```
func (t *ChainCode) Init(stub *shim.ChaincodeStub, function string, args []string) ([]byte, error) {
       err := stub.CreateTable("TempTable", []*shim.ColumnDefinition{
        &shim.ColumnDefinition{"Column1", shim.ColumnDefinition_BYTES, true},
        &shim.ColumnDefinition{"Column2", shim.ColumnDefinition_INT64, false}})
    if err != nil {
        return nil, errors.New("Failed creating TempTable")
    }
       return nil, nil
}
```

Output when i deploy the chaincode:

```
2016/04/11 14:24:20 Peer address: 127.0.0.1:30303
2016/04/11 14:24:20 os.Args returns: [./chaincodetest]
2016/04/11 14:24:20 Chaincode ID: chaincodetest
2016/04/11 14:24:20 Registering.. sending REGISTER
2016/04/11 14:24:20 []Received message REGISTERED from shim
2016/04/11 14:24:20 []Handling ChaincodeMessage of type: REGISTERED(state:created)
2016/04/11 14:24:20 Received REGISTERED, ready for invocations
```

Output after i called the Init function 

```
2016/04/11 14:31:37 Peer address: 127.0.0.1:30303
2016/04/11 14:31:37 os.Args returns: [./chaincodetest]
2016/04/11 14:31:37 Chaincode ID: chaincodetest
2016/04/11 14:31:37 Registering.. sending REGISTER
2016/04/11 14:31:37 []Received message REGISTERED from shim
2016/04/11 14:31:37 []Handling ChaincodeMessage of type: REGISTERED(state:created)
2016/04/11 14:31:37 Received REGISTERED, ready for invocations
2016/04/11 14:31:47 [chaincodetest]Received message INIT from shim
2016/04/11 14:31:47 [chaincodetest]Handling ChaincodeMessage of type: INIT(state:established)
2016/04/11 14:31:47 Entered state init
2016/04/11 14:31:47 [chaincodetest]Received INIT, initializing chaincode
2016/04/11 14:31:47 Setting up the HoldingsTable
2016/04/11 14:31:47 [chaincodetest]Sending GET_STATE
2016/04/11 14:31:47 [chaincodetest]Received message ERROR from shim
2016/04/11 14:31:47 [chaincodetest]Handling ChaincodeMessage of type: ERROR(state:init)
2016/04/11 14:31:47 [chaincodetest]before send
2016/04/11 14:31:47 [chaincodetest]after send
2016/04/11 14:31:47 [chaincodetest]Error received from validator ERROR, communicated(state:established)
2016/04/11 14:31:47 [chaincodetest]GetState received error ERROR
2016/04/11 14:31:47 [chaincodetest]Inside putstate, isTransaction = true
2016/04/11 14:31:47 [chaincodetest]Sending PUT_STATE
2016/04/11 14:31:47 [chaincodetest]Received message RESPONSE from shim
2016/04/11 14:31:47 [chaincodetest]Handling ChaincodeMessage of type: RESPONSE(state:established)
Error starting Simple chaincode: Error handling message: [chaincodetest]Chaincode handler FSM cannot handle message (RESPONSE) with payload size (0) while in state: established%
```

Is this a known issue or did i do something wrong?

**Comment by muralisrini** [comment]
@novusopt : We need to track down the first error 

2016/04/11 14:31:47 [chaincodetest]Received INIT, initializing chaincode
2016/04/11 14:31:47 Setting up the HoldingsTable
2016/04/11 14:31:47 [chaincodetest]Sending GET_STATE
2016/04/11 14:31:47 [chaincodetest]Received message ERROR from shim

Can you send entire logs from both sides ? And if its ok with you, the chaincode ?

**Comment by srderson** [comment]
@muralisrini We test the following similar chaincode https://github.com/hyperledger/fabric/blob/master/core/behave_chaincode/go/table/table.go (in that it creates tables in the Init function) in the BDD tests and it was not encountering this issue. One difference is that we're not testing this one with security enabled. Not sure if that would make a difference.

**Comment by novusopt** [comment]
@muralisrini 
Sure

chaincode

```
package main

import (

    "fmt"
    "github.com/hyperledger/fabric/core/chaincode/shim"
    "errors"
)

type ChainCode struct {
}


func (t *ChainCode) Init(stub *shim.ChaincodeStub, function string, args []string) ([]byte, error) {

    err := stub.CreateTable("TempTable", []*shim.ColumnDefinition{
        &shim.ColumnDefinition{"Column1", shim.ColumnDefinition_BYTES, true},
        &shim.ColumnDefinition{"Column2", shim.ColumnDefinition_INT64, false}})

    if err != nil {
        return nil, errors.New("Failed creating TempTable")
    }

    return nil, nil
}


func (t *ChainCode) Invoke(stub *shim.ChaincodeStub, function string, args []string) ([]byte, error) {

    //nothing to do

    return nil, nil
}

func (t *ChainCode) Query(stub *shim.ChaincodeStub, function string, args []string) ([]byte, error) {

    //nothing to do

    return nil, nil
}

func main() {
    err := shim.Start(new(ChainCode))
    if err != nil {
        fmt.Printf("Error starting Simple chaincode: %s", err)
    }

}
```

validation peer log output

```
15:39:20.380 [crypto] main -> INFO 001 Log level recognized 'info', set to INFO
15:39:20.380 [logging] LoggingInit -> DEBU 002 Setting default logging level to DEBUG for command 'peer'
15:39:20.381 [eventhub_producer] AddEventType -> DEBU 003 registering block
15:39:20.381 [eventhub_producer] AddEventType -> DEBU 004 registering register
15:39:20.381 [main] serve -> INFO 005 Running in chaincode development mode
15:39:20.381 [main] serve -> INFO 006 Set consensus to NOOPS and user starts chaincode
15:39:20.381 [main] serve -> INFO 007 Disable loading validity system chaincode
15:39:20.381 [main] serve -> INFO 008 Security enabled status: true
15:39:20.381 [main] serve -> INFO 009 Privacy enabled status: true
15:39:20.381 [main] serve -> DEBU 00a Running as validating peer - installing consensus noops
15:39:20.381 [eventhub_producer] start -> INFO 00b event processor started
15:39:20.381 [peer] NewPeerWithHandler -> DEBU 00c Registering validator with enroll ID: vp
15:39:20.381 [crypto] NewPeerWithHandler -> INFO 00d Registering validator [vp] with name [vp]...
15:39:20.382 [crypto] register -> ERRO 00e Failed registering [vp]: [Already registered.]
15:39:20.382 [crypto] RegisterValidator -> ERRO 00f Failed registering [vp]: [Already registered.]
15:39:20.382 [crypto] NewPeerWithHandler -> INFO 010 Registering vlidator [vp] with name [vp]...done. Already registered or initiliazed.
15:39:20.382 [crypto] NewPeerWithHandler -> INFO 011 Registering validator [vp] with name [vp]...done!
15:39:20.382 [peer] NewPeerWithHandler -> DEBU 012 Initializing validator with enroll ID: vp
15:39:20.382 [crypto] NewPeerWithHandler -> INFO 013 Initializing validator [vp]...
15:39:20.399 [crypto] NewPeerWithHandler -> INFO 014 Initializing validator [vp]...done!
15:39:20.400 [db] createDBIfDBPathEmpty -> DEBU 015 Is db path [/var/hyperledger/production/db] empty [false]
15:39:20.405 [state] NewState -> INFO 016 Initializing state implementation [buckettree]
15:39:20.405 [buckettree] initConfig -> INFO 017 configs passed during initialization = map[string]interface {}{"numBuckets":10009, "maxGroupingAtEachLevel":10}
15:39:20.405 [buckettree] initConfig -> INFO 018 Initializing bucket tree state implemetation with configurations &{maxGroupingAtEachLevel:10 lowestLevel:5 levelToNumBucketsMap:map[4:1001 3:101 2:11 1:2 0:1 5:10009] hashFunc:0x460e860}
15:39:20.405 [buckettree] computeCryptoHash -> DEBU 019 Appending crypto-hash for child bucket = [level=[1], bucketNumber=[1]]
15:39:20.405 [buckettree] computeCryptoHash -> DEBU 01a Propagating crypto-hash of single child node for bucket = [level=[0], bucketNumber=[1]]
15:39:20.405 [peer] chatWithPeer -> DEBU 01b Starting up the first peer
15:39:20.406 [chaincode] NewChaincodeSupport -> INFO 01c Chaincode support using peerAddress: 0127.0.0.1:30303
15:39:20.406 [rest] StartOpenchainRESTServer -> INFO 01d Initializing the REST service...
15:39:20.406 [main] serve -> INFO 01e Starting peer with id=name:"vp0" , network id=dev, address=0127.0.0.1:30303, discovery.rootnode=, validator=true
15:39:38.028 [chaincode] HandleChaincodeStream -> DEBU 01f Current context deadline = 0001-01-01 00:00:00 +0000 UTC, ok = false
15:39:38.029 [chaincode] processStream -> DEBU 020 []Received message REGISTER from shim
15:39:38.029 [chaincode] HandleMessage -> DEBU 021 []Handling ChaincodeMessage of type: REGISTER in state created
15:39:38.029 [chaincode] beforeRegisterEvent -> DEBU 022 Received REGISTER in state created
15:39:38.029 [chaincode] registerHandler -> DEBU 023 registered handler complete for chaincode chaincodetest
15:39:38.029 [chaincode] beforeRegisterEvent -> DEBU 024 Got REGISTER for chaincodeID = name:"chaincodetest" , sending back REGISTERED
15:39:38.029 [chaincode] notifyDuringStartup -> DEBU 025 nothing to notify (dev mode ?)
15:40:14.557 [crypto] Login -> INFO 026 Registering client [user1] with name [user1]...
15:40:14.558 [crypto] Login -> INFO 027 Registering client [user1] with name [user1]...done. Already registered or initiliazed.
15:40:14.558 [crypto] Login -> INFO 028 Registering client [user1] with name [user1]...done!
15:40:18.429 [devops] Deploy -> DEBU 029 Initializing secure devops using context user1
15:40:18.429 [crypto] Deploy -> INFO 02a Initializing client [user1]...
15:40:18.642 [crypto] Deploy -> INFO 02b Initializing client [user1]...done!
15:40:18.642 [devops] Deploy -> DEBU 02c Creating secure transaction chaincodetest
15:40:18.643 [devops] Deploy -> DEBU 02d Sending deploy transaction (chaincodetest) to validator
15:40:18.643 [peer] sendTransactionsToThisPeer -> DEBU 02e Marshalling transaction CHAINCODE_DEPLOY to send to self
15:40:18.643 [peer] sendTransactionsToThisPeer -> DEBU 02f Sending message CHAIN_TRANSACTION with timestamp seconds:1460382018 nanos:643853681  to self
15:40:18.643 [peer] handleChat -> DEBU 030 Current context deadline = 0001-01-01 00:00:00 +0000 UTC, ok = false
15:40:18.644 [consensus/noops] newNoops -> DEBU 031 Creating a NOOPS object
15:40:18.646 [consensus/noops] newNoops -> INFO 032 NOOPS consensus type = *noops.Noops
15:40:18.646 [consensus/noops] newNoops -> INFO 033 NOOPS block size = 500
15:40:18.646 [consensus/noops] newNoops -> INFO 034 NOOPS block timeout = 1s
15:40:18.646 [consensus/handler] doChainTransaction -> DEBU 035 Verifying transaction signature chaincodetest
15:40:18.646 [consensus/handler] SendMessage -> DEBU 036 Sending to stream a message of type: RESPONSE
15:40:18.646 [peer] SendMessage -> DEBU 037 Sending message to stream of type: RESPONSE
15:40:18.647 [peer] func1 -> DEBU 038 Received RESPONSE message as expected, will wait for EOF
15:40:18.647 [consensus/noops] RecvMsg -> DEBU 039 Handling Message of type: CHAIN_TRANSACTION
15:40:18.647 [consensus/noops] broadcastConsensusMsg -> DEBU 03a Broadcasting CONSENSUS
15:40:18.647 [consensus/noops] RecvMsg -> DEBU 03b Sending to channel tx uuid: %!(EXTRA string=chaincodetest)
15:40:18.647 [peer] handleChat -> DEBU 03c Received EOF, ending Chat
15:40:18.647 [peer] func1 -> DEBU 03d Received EOF
15:40:18.647 [crypto] CloseClient -> INFO 03e Closing client [user1]...
15:40:19.650 [consensus/noops] handleChannels -> DEBU 03f Process block due to time
15:40:19.650 [consensus/noops] processTransactions -> DEBU 040 Starting TX batch with timestamp: seconds:1460382019 nanos:650416993
15:40:19.650 [consensus/noops] processTransactions -> DEBU 041 Executing batch of 1 transactions with timestamp seconds:1460382019 nanos:650416993
15:40:19.651 [chaincode] DeployChaincode -> DEBU 042 user runs chaincode, not deploying chaincode
15:40:19.651 [state] TxBegin -> DEBU 043 txBegin() for txUuid [chaincodetest]
15:40:19.651 [chaincode] LaunchChaincode -> DEBU 044 Container not in READY state(established)...send init/ready
15:40:19.651 [chaincode] initOrReady -> DEBU 045 sending INIT
15:40:19.651 [chaincode] setChaincodeSecurityContext -> DEBU 046 setting chaincode security context...
15:40:19.651 [chaincode] setChaincodeSecurityContext -> DEBU 047 setting chaincode security context. Transaction different from nil
15:40:19.651 [chaincode] setChaincodeSecurityContext -> DEBU 048 setting chaincode security context. Metadata []
15:40:19.652 [chaincode] processStream -> DEBU 049 [chaincod]Move state message INIT
15:40:19.652 [chaincode] HandleMessage -> DEBU 04a [chaincod]Handling ChaincodeMessage of type: INIT in state established
15:40:19.652 [chaincode] beforeInitState -> DEBU 04b Before state established.. notifying waiter that we are up
15:40:19.652 [chaincode] notifyDuringStartup -> DEBU 04c nothing to notify (dev mode ?)
15:40:19.652 [chaincode] enterInitState -> DEBU 04d [chaincod]Entered state init
15:40:19.653 [chaincode] processStream -> DEBU 04e [chaincod]Received message GET_STATE from shim
15:40:19.653 [chaincode] HandleMessage -> DEBU 04f [chaincod]Handling ChaincodeMessage of type: GET_STATE in state init
15:40:19.653 [chaincode] afterGetState -> DEBU 050 [chaincod]Received GET_STATE, invoking get state from ledger
15:40:19.653 [chaincode] filterError -> DEBU 051 Ignoring NoTransitionError: no transition
15:40:19.653 [buckettree] newDataKey -> DEBU 052 Enter - newDataKey. chaincodeID=[chaincodetest], key=[9TempTable]
15:40:19.653 [buckettree] newDataKey -> DEBU 053 Exit - newDataKey=[bucketKey=[level=[5], bucketNumber=[2975]], compositeKey=[chaincodetest9TempTable]]
15:40:19.653 [chaincode] encryptOrDecrypt -> DEBU 054 [chaincod]Payload before encrypt/decrypt: []
15:40:19.653 [chaincode] encryptOrDecrypt -> DEBU 055 [chaincod]Payload after encrypt/decrypt: []
15:40:19.653 [chaincode] func1 -> ERRO 056 [chaincod]Got error (Decryption failed.) while decrypting. Sending ERROR
15:40:19.653 [chaincode] 1 -> DEBU 057 [chaincod]handleGetState serial send ERROR
15:40:19.653 [chaincode] processStream -> DEBU 058 [chaincod]Received message PUT_STATE from shim
15:40:19.653 [chaincode] HandleMessage -> DEBU 059 [chaincod]Handling ChaincodeMessage of type: PUT_STATE in state init
15:40:19.653 [chaincode] afterPutState -> DEBU 05a Received PUT_STATE in state busyinit, invoking put state to ledger
15:40:19.654 [chaincode] func1 -> DEBU 05b [chaincod]state is busyinit
15:40:19.654 [chaincode] encryptOrDecrypt -> DEBU 05c [chaincod]Payload before encrypt/decrypt: [10 9 84 101 109 112 84 97 98 108 101 18 13 10 7 67 111 108 117 109 110 49 16 5 24 1 18 11 10 7 67 111 108 117 109 110 50 16 2]
15:40:19.654 [chaincode] encryptOrDecrypt -> DEBU 05d [chaincod]Payload after encrypt/decrypt: [186 73 77 28 18 75 217 135 75 28 255 157 48 228 140 147 84 51 5 179 38 138 135 104 195 70 100 221 238 48 219 93 191 58 113 133 220 95 123 144 170 140 173 67 104 121 53 74 19 168 135 5 101 108 14 172 18 122 148 123 143 140 108 149 54 31 240 138 71 192 99 123 120 25 140 78 129 145 204 242 175 32 13 76 221 101 204 176 191 150 17]
15:40:19.654 [state] Set -> DEBU 05e set() chaincodeID=[chaincodetest], key=[9TempTable], value=[[]byte{0xba, 0x49, 0x4d, 0x1c, 0x12, 0x4b, 0xd9, 0x87, 0x4b, 0x1c, 0xff, 0x9d, 0x30, 0xe4, 0x8c, 0x93, 0x54, 0x33, 0x5, 0xb3, 0x26, 0x8a, 0x87, 0x68, 0xc3, 0x46, 0x64, 0xdd, 0xee, 0x30, 0xdb, 0x5d, 0xbf, 0x3a, 0x71, 0x85, 0xdc, 0x5f, 0x7b, 0x90, 0xaa, 0x8c, 0xad, 0x43, 0x68, 0x79, 0x35, 0x4a, 0x13, 0xa8, 0x87, 0x5, 0x65, 0x6c, 0xe, 0xac, 0x12, 0x7a, 0x94, 0x7b, 0x8f, 0x8c, 0x6c, 0x95, 0x36, 0x1f, 0xf0, 0x8a, 0x47, 0xc0, 0x63, 0x7b, 0x78, 0x19, 0x8c, 0x4e, 0x81, 0x91, 0xcc, 0xf2, 0xaf, 0x20, 0xd, 0x4c, 0xdd, 0x65, 0xcc, 0xb0, 0xbf, 0x96, 0x11}]
15:40:19.654 [buckettree] newDataKey -> DEBU 05f Enter - newDataKey. chaincodeID=[chaincodetest], key=[9TempTable]
15:40:19.654 [buckettree] newDataKey -> DEBU 060 Exit - newDataKey=[bucketKey=[level=[5], bucketNumber=[2975]], compositeKey=[chaincodetest9TempTable]]
15:40:19.654 [chaincode] func1 -> DEBU 061 [chaincod]Completed PUT_STATE. Sending RESPONSE
15:40:19.654 [chaincode] 1 -> DEBU 062 [chaincod]enterBusyState trigger event RESPONSE
15:40:19.654 [chaincode] processStream -> DEBU 063 [chaincod]Move state message RESPONSE
15:40:19.654 [chaincode] HandleMessage -> DEBU 064 [chaincod]Handling ChaincodeMessage of type: RESPONSE in state busyinit
15:40:19.654 [chaincode] enterInitState -> DEBU 065 [chaincod]Entered state init
15:40:19.654 [chaincode] processStream -> DEBU 066 [chaincod]sending state message RESPONSE
15:40:19.654 [chaincode] processStream -> DEBU 067 [chaincod]Received message ERROR from shim
15:40:19.654 [chaincode] processStream -> DEBU 068 Got error: [chaincodetest]Chaincode handler FSM cannot handle message (RESPONSE) with payload size (0) while in state: established
15:40:19.654 [chaincode] HandleMessage -> DEBU 069 [chaincod]Handling ChaincodeMessage of type: ERROR in state init
15:40:19.654 [chaincode] enterEndState -> DEBU 06a [chaincod]Entered state end
15:40:19.654 [chaincode] notify -> DEBU 06b notifying Uuid:chaincodetest
15:40:19.654 [chaincode] deregisterHandler -> DEBU 06c Deregister handler: chaincodetest
15:40:19.654 [chaincode] deregisterHandler -> DEBU 06d Deregistered handler with key: chaincodetest
15:40:19.654 [chaincode] processStream -> DEBU 06e Received EOF, ending chaincode support stream, EOF
15:40:19.654 [chaincode] deregisterHandler -> DEBU 06f Deregister handler: chaincodetest
15:40:19.654 [chaincode] LaunchChaincode -> DEBU 070 sending init failed(Error initializing container chaincodetest: [chaincodetest]Chaincode handler FSM cannot handle message (RESPONSE) with payload size (0) while in state: established)
15:40:19.655 [container] lockContainer -> DEBU 071 waiting for container(dev-vp0-chaincodetest) lock
15:40:19.655 [container] lockContainer -> DEBU 072 got container (dev-vp0-chaincodetest) lock
15:40:19.655 [container] newDockerClient -> INFO 073 Creating VM with endpoint: unix:///var/run/docker.sock
15:40:19.655 [container] stopInternal -> DEBU 074 Stop container dev-vp0-chaincodetest(Post http://unix.sock/containers/dev-vp0-chaincodetest/stop?t=0: dial unix /var/run/docker.sock: connect: no such file or directory)
15:40:19.655 [container] stopInternal -> DEBU 075 Kill container dev-vp0-chaincodetest (Post http://unix.sock/containers/dev-vp0-chaincodetest/kill: dial unix /var/run/docker.sock: connect: no such file or directory)
15:40:19.656 [container] stopInternal -> DEBU 076 Remove container dev-vp0-chaincodetest (dial unix /var/run/docker.sock: connect: no such file or directory)
15:40:19.656 [container] unlockContainer -> DEBU 077 container lock deleted(dev-vp0-chaincodetest)
15:40:19.656 [chaincode] LaunchChaincode -> DEBU 078 sending init completed
15:40:19.656 [chaincode] LaunchChaincode -> DEBU 079 LaunchChaincode complete
15:40:19.656 [state] TxFinish -> DEBU 07a txFinish() for txUuid [chaincodetest], txSuccessful=[false]
15:40:19.656 [state] GetHash -> DEBU 07b Enter - GetHash()
15:40:19.656 [buckettree] ComputeCryptoHash -> DEBU 07c Enter - ComputeCryptoHash()
15:40:19.656 [buckettree] ComputeCryptoHash -> DEBU 07d Returing existing crypto-hash as recomputation not required
15:40:19.656 [state] GetHash -> DEBU 07e Exit - GetHash()
15:40:19.656 [consensus/noops] processTransactions -> DEBU 07f Committing TX batch with timestamp: seconds:1460382019 nanos:650416993
15:40:19.656 [state] GetHash -> DEBU 080 Enter - GetHash()
15:40:19.656 [buckettree] ComputeCryptoHash -> DEBU 081 Enter - ComputeCryptoHash()
15:40:19.656 [buckettree] ComputeCryptoHash -> DEBU 082 Returing existing crypto-hash as recomputation not required
15:40:19.656 [state] GetHash -> DEBU 083 Exit - GetHash()
15:40:19.656 [indexes] addIndexDataForPersistence -> DEBU 084 Indexing block number [19] by hash = [ec1c1ed99586f6409df37531dd37af74a9d934eab42f21c000964be1c011ad4129b60021dd16082983dad881234519d0a8fda6865491efb3223d401f523e29c9]
15:40:19.656 [state] AddChangesForPersistence -> DEBU 085 state.addChangesForPersistence()...start
15:40:19.656 [state] AddChangesForPersistence -> DEBU 086 Adding state-delta corresponding to block number[19]
15:40:19.656 [state] AddChangesForPersistence -> DEBU 087 Not deleting previous state-delta. Block number [19] is smaller than historyStateDeltaSize [500]
15:40:19.656 [state] AddChangesForPersistence -> DEBU 088 state.addChangesForPersistence()...finished
15:40:19.656 [ledger] resetForNextTxGroup -> DEBU 089 resetting ledger state for next transaction batch
15:40:19.656 [buckettree] ClearWorkingSet -> DEBU 08a Enter - ClearWorkingSet()
15:40:19.656 [ledger] sendProducerBlockEvent -> ERRO 08b Error unmarshalling deployment transaction for block event: proto: protos.ChaincodeDeploymentSpec: wiretype end group for non-group
15:40:19.656 [consensus/noops] getBlockData -> DEBU 08c Preparing to broadcast with block number 20
15:40:19.656 [consensus/noops] getBlockData -> DEBU 08d Got the delta state of block number 20
15:40:19.656 [consensus/noops] notifyBlockAdded -> DEBU 08e Broadcasting Message_SYNC_BLOCK_ADDED to non-validators

```

**Comment by muralisrini** [comment]
thanks, @novusopt . I see this as being the source of the problem 
15:40:19.653 [chaincode] encryptOrDecrypt -> DEBU 054 [chaincod]Payload before encrypt/decrypt: []
15:40:19.653 [chaincode] encryptOrDecrypt -> DEBU 055 [chaincod]Payload after encrypt/decrypt: []
15:40:19.653 [chaincode] func1 -> ERRO 056 [chaincod]Got error (Decryption failed.) while decrypting. Sending ERROR

I think you have to deploy with security enabled 

CORE_SECURITY_ENABLED=true CORE_SECURITY_PRIVACY=true ./fabric chaincode deploy -n chaincodetest -u user1 -c '{"Function":"init", "Args": ["arg1"]}'

To start afresh, delete persistent stores (but leave the hyperledger dir around)
rm -r /var/hyperledger/\* 

We will try the above too soon.

**Comment by novusopt** [comment]
@muralisrini i deployed with security enabled but i still get the same error.

**Comment by muralisrini** [comment]
Will recreate over next two days

**Comment by muralisrini** [comment]
@novusopt : I was able to recreate and have some idea as to what's causing this. I need to investigate this further. Will update when I know more

**Comment by novusopt** [comment]
@muralisrini Thanks!

**Comment by cp68** [comment]
Running into the same issue as above. Do we know the cause? Can I help?

**Comment by muralisrini** [comment]
@cp68 :  absolutely welcome help ... thanks!

I believe this is what's happening : 

At a high level, CreateTable uses the GetState internally to check for some state. With privacy enabled, state obtained through GetState is decrypted on the peer side before sending it to the chaincode.  The bug is that we are trying to decrypt the results of empty GetState which is treated by the peer side as an error. While this error does not affect the caller of CreateTable (it treats the error as not-exist error) it does play havoc on the chaincode handling.

What we should really do is attempt decrypt from GetState only if the state exists, otherwise return "not exist" error to chaincode. CreateTable is actually looking for some error anyway and won't be 

Do let me know if you need code pointers and I can recreate and update (will need a bit of time).

**Comment by muralisrini** [comment]
@cp68 - #1247 seems related to this as well.

**Comment by cp68** [comment]
Created pull request #1373. @muralisrini - Can you test? I wasn't able to consistently replicate the issue.

**Comment by novusopt** [comment]
@cp68: i tested your pull request, i get still the same error

**Comment by cp68** [comment]
After getting the latest master, I cannot replicate. I get the following when trying to deploy the chaincode mentioned above. I have security and privacy enabled. This seems to indicate that the problem no longer exists.

.....
20:44:19.947 [chaincode] encryptOrDecrypt -> DEBU 13f [chaincod]Payload before encrypt/decrypt: []
20:44:19.947 [chaincode] encryptOrDecrypt -> DEBU 140 [chaincod]Payload after encrypt/decrypt: []
20:44:19.947 [chaincode] func1 -> DEBU 141 [chaincod]Got state. Sending RESPONSE

**Comment by adecaro** [comment]
Hi @muralisrini, @cp68, you don't see the error anymore because I patched the crypto layer to return just nil without errors when trying do decrypt a nil ciphertext. 

I have introduced this patch to quickly fix #1247. Actually, I think the right way to address this is that proposed by @cp68 in his PR. There, it is avoided to pass a nil ciphertext to crypto layer.

**Comment by srderson** [comment]
Closing as this should be resolved now.
